pylint/checkers/logging.py

```diff
@@ -215,6 +229,23 @@ def is_logger_class():
                 return
         self._check_log_method(node, name)
 
+    def _is_lazy_needed(self, level):
+        level = level.lower()
+
+        if "all" not in self._check_levels:
```

Comment:
actually i would love a linter that detected this:  `short-circuit-readability`.   Inverting the logic of the code allows the "return true" to happen first, and the rest to be unintended.